PythonVisualizations/Array.py

```diff
@@ -353,12 +387,18 @@ def delete(self, item):
 
     def remove(self, val):
         self.startAnimations()
+        
+        # draw an index pointing to the last item in the list
+        indexDisplay = self.createIndex(len(self.list)-1, 'nItems',level = -1, color = 'black')
+                
         callEnviron = self.createCallEnvironment(
             self.removeCode.strip(), self.removeCodeSnippets)
-
+        
+        callEnviron |= set(indexDisplay)         

```

Comment:
There's no need to recreate the `nItems` index in each operation.  Use the one built in the `display()` method.